Source Packages/java/edu/slu/tpen/servlet/ProjectServlet.java

```diff
@@ -65,7 +65,7 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
                     String redirectURL = req.getContextPath() + "/newberryTrans.html?projectID=" + projID;
                     resp.sendRedirect(redirectURL);
                 } else {
-                    projID = Integer.parseInt(req.getPathInfo().substring(1).replace("/", ""));
+                    projID = Integer.parseInt(req.getPathInfo().substring(1).replace("/", "").replace("manifest.json",""));
```

Comment:
I don't think parseInt() cares about anything after the number, but this shouldn't hurt either. I mean `435/manifest.json` and `435/TiggersAreOnly1` should both work.

Reply:
Yeah, but then it isn't a pattern, it is a wildcard.  I want it to NOT work when you ask for it incorrectly.  So I disagree and think `http://paleo.rerum.io/TPEN-NL/project/4080/getFirstPage` should fail to return anything.

Reply:
I agree. I tested this out after I commented. In Java, Integer.parseInt() needs a whole number string and nothing else, which I didn't know. In JavaScript, `parseInt("21gimli")` is `21` which is crazy, but means that if this behaved that way, `/getFirstPage` wouldn't break anything.

I like it how it is. Good work.